exec/java-exec/src/main/resources/rest/storage/list.ftl

```diff
@@ -17,79 +17,280 @@
+                </button>
+                <button type="button" class="btn btn-primary" name="${plugin.getName()}" data-toggle="modal"
+                        data-target="#pluginsModal">
+                  Export
+                </button>
               </td>
             </tr>
           </#if>
         </#list>
       </tbody>
     </table>
   </div>
-  <div class="page-header">
+
+
+  <#-- Modal window for exporting plugin config (including group plugins modal) -->
+  <div class="modal fade" id="pluginsModal" tabindex="-1" role="dialog" aria-labelledby="exportPlugin" aria-hidden="true">
+    <div class="modal-dialog modal-sm" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+          <h4 class="modal-title" id="exportPlugin">Plugin config</h4>
+        </div>
+        <div class="modal-body">
+          <div id="format" style="display: inline-block; position: relative;">
+            <label for="format">Format</label>
+            <div class="radio">
+              <label>
+                <input type="radio" name="format" id="json" value="json" checked="checked">
+                JSON
+              </label>
+            </div>
+            <div class="radio">
+              <label>
+                <input type="radio" name="format" id="hocon" value="conf">
+                HOCON
+              </label>
+            </div>
+          </div>
+
+          <div id="plugin-set" class="" style="display: inline-block; position: relative; float: right;">
+            <label for="format">Plugin group</label>
+            <div class="radio">
+              <label>
+                <input type="radio" name="group" id="all" value="all" checked="checked">
+                ALL
+              </label>
+            </div>
+            <div class="radio">
+              <label>
+                <input type="radio" name="group" id="enabled" value="enabled">
+                ENABLED
+              </label>
+            </div>
+            <div class="radio">
+              <label>
+                <input type="radio" name="group" id="disabled" value="disabled">
+                DISABLED
+              </label>
+            </div>
+          </div>
+        </div>
+
+        <div class="modal-footer">
+          <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+          <button type="button" id="export" class="btn btn-primary">Export</button>
+        </div>
+      </div>
+    </div>
   </div>
-  <div>
-    <h4>New Storage Plugin</h4>
-    <form class="form-inline" id="newStorage" role="form" action="/" method="GET">
-      <div class="form-group">
-        <input type="text" class="form-control" id="storageName" placeholder="Storage Name">
+  <#-- Modal window for exporting plugin config (including group plugins modal) -->
+
+  <#-- Modal window for creating plugin -->
```

Comment:
Not a must, but I recommend moving **all** modals to `exec/java-exec/src/main/resources/rest/alertModals.ftl`
This will keep everything the code to minimal.

Reply:
I started do that, but then rethought it.
Then `alertModals.ftl` looks a bit messy and other pages (options, profile, query) will include additional not necessary code. The pages become larger.

I think it makes sense to factor out common modals and scripts. These modals are specific and used only on one page. So it is fine to keep them in the appropriate FTL page now. If so, it can be changed in future.